software/controller/lib/core/alarm.cpp

```diff
@@ -0,0 +1,102 @@
+#include <stdint.h>
+#include "vars.h"
+#include "debug.h"
+#include "hal.h"
+#include "alarm.h"
+#include "comms.h"
+#include "commfail_alarm.h"
+
+// MAX_CONTROLLER_ALARMS 
+#define MAX_CONTROLLER_ALARMS 2
+
+// structure definition for MAX_CONTROLLER_ALARMS 
+alarm_cb_t alarms[MAX_CONTROLLER_ALARMS];
+
+// this value is reset to 0 after MAX_CONTROLLER_ALARMS 
+static uint16_t alarm_check_counter;
+
+// There should be a packet from GUI for every 30ms. Assuming something went
+// wrong, we wait for 50ms before raising a communication failure alarm
+static constexpr Duration COMMUNICATION_TIMEOUT = milliseconds(10000);
+static DebugUInt32 dbg_alarm_check("alarm_check",
+                                      "alarm_check",
+                                      0);
+uint32_t alarm_check;
+
+extern Time last_rx;
+extern Time comms_get_lastrx_time(void);
+
+static bool is_time_to_raise_alarm() {
+    return Hal.now() - comms_get_lastrx_time() > COMMUNICATION_TIMEOUT;
+}
+
+void alarm_init(void)
+{
+    // for COMMUNICATION_CHECK_ALARM 1
+    alarms[COMM_CHECK_ALARM].timestamp = Hal.now();
+    alarms[COMM_CHECK_ALARM].callthis = commfail_alarm_cb;
+    alarms[COMM_CHECK_ALARM].priority = ALARM_HIGH_PRIORITY;
+    alarms[COMM_CHECK_ALARM].triggered = 0;
+    alarms[COMM_CHECK_ALARM].alarm_pause = 0;// 0 not pausable, 1 pausable
```

Comment:
If there is meaning other than `true` and `false` (for which `bool` would be the appropriate type), then use a scoped enum class and define those values. Or maybe then the member variables should just be called `alarm_pausable` and `audio_pausable`.